### ISSUE TITLE
Multiple issues with keyboard navigation (menu and inputtext)

### ISSUE THREAD

**Issue by kudaba** [ocornut/imgui#3563]
**Version/Branch of Dear ImGui:**

Version: 1.77 (it's a bit behind but I didn't see any differences in the nav code)
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_win32.cpp + custom renderer
Compiler: MSVC
Operating System: Windows

**My Issue/Question:**

I noticed some odd behaviors when trying to create a filterable popup menu.
1. If you use BeginMenuBar multiple times, the first usage can react to the left/right navigation before submenus that were created in the second instance.
    * My local fix was to re-architect my code to only invoke it once per window.
```cpp
		ImGui::BeginMenuBar();
		if (ImGui::BeginMenu("First"))
		{
			ImGui::MenuItem("Normal");
			ImGui::EndMenu();
		}
		ImGui::EndMenuBar();

		ImGui::BeginMenuBar();
		if (ImGui::BeginMenu("Second"))
		{
			if (ImGui::BeginMenu("Can't enter with right"))
			{
				ImGui::MenuItem("Can't leave with left");
				ImGui::EndMenu();
			}
			ImGui::EndMenu();
		}
		ImGui::EndMenuBar();
```
2. Even when using BeginMenuBar once, using the left navigation to leave a sub menus has a bit odd behavior.
![menu_nav_left](https://user-images.githubusercontent.com/22896740/97401454-16c02780-18ae-11eb-8bf9-5700ca72b53e.gif)
3. If you are making a popup menu using BeginPopup, the left navigation will not return you to the bottom level menu. You can close the popup using EndMenu instead of EndPopup but that feels a bit odd.
    * My local fix was to add BeginPopupMenu/EndPopupMenu wrappers that do that but look more correct.
```cpp
		if (ImGui::IsMouseClicked(1))
			ImGui::OpenPopup("test");

		if (ImGui::BeginPopup("test"))
		{
			if (ImGui::BeginMenu("SubMenu"))
			{
				ImGui::MenuItem("Can't leave with left");
				ImGui::EndMenu();
			}
			ImGui::EndPopup();
		}
```
4. When using keyboard navigation with InputText areas, you can use the arrow key to exist an input box, but when you navigate back you just get the selection rectangle. It would feel better if if single line text boxes would automatically take keyboard focus, or have an option somewhere to enable that behavior.
    * My local fix here is to call SetKeyboardFocusHere the same frame the navigation id changes to the text box.
![input_nav](https://user-images.githubusercontent.com/22896740/97399975-8385f280-18ab-11eb-8530-ec4f7790d020.gif)

**Comment by ocornut** [comment]
> It would feel better if if single line text boxes would automatically take keyboard focus, or have an option somewhere to enable that behavior. 

Yes you can do e.g. `if (g.NavId == GetID("MyTextInput")) ActivateItem(g.NavID)` for that second part.
